app/javascript/crayons/Button/Button.jsx

```diff
@@ -0,0 +1,110 @@
+import { h } from 'preact';
+import PropTypes from 'prop-types';
+import { defaultChildrenPropTypes } from '../../src/components/common-prop-types';
+
+function getAdditionalClassNames({
+  variant,
+  className,
+  icon,
+  disabled,
+  children,
+}) {
+  let additionalClassNames = '';
+
+  if (variant && variant.length > 0 && variant !== 'primary') {
+    additionalClassNames += ` crayons-btn--${variant}`;
+  }
+
+  if (icon) {
+    additionalClassNames +=
+      children.length > 0
+        ? ' crayons-btn--icon-left'
+        : ' crayons-btn--icon-alone';
+  }
+
+  if (disabled) {
+    additionalClassNames += ' crayons-btn--disabled';
+  }
+
+  if (className && className.length > 0) {
+    additionalClassNames += ` ${className}`;
+  }
+
+  return additionalClassNames;
+}
+
+export const Button = ({
+  children,
+  variant = 'primary',
+  as = 'button',
```

Comment:
mmm, I saw you note about using `as` but as I see it's immediately assigned to `ComponentName` as a const, what if we call it `htmlElement` instead of `as` or `componentName` ?

Reply:
The funny thing about JSX is a component name has to be Pascal cased. So it does look odd setting `const ComponentName = as;`. Instead of `htmlElement`, it actually makes sense to call it `tagName` as that's what it really is, https://developer.mozilla.org/en-US/docs/Web/API/Element/tagName.